esptool.py

```diff
@@ -3075,6 +3115,92 @@ def make_image(args):
     image.save(args.output)
 
 
+def merge_image(args):
+    import espsecure
+
+    # Helper classes
+    class Bin(object):
+        def __init__(self, addr, file):
+            self.addr = addr
+            self.file = file
+            self.file_name = os.path.basename(file.name)
+            self.size = self.get_size()
+
+        def get_size(self):
+            self.file.seek(0, 2)  # seek to end
+            size = self.file.tell()
+            self.file.seek(0)
+            return size
+
+    class MultiBin(object):
+        def __init__(self, name, output_folder='.'):
+            self.name = name
+            self.output_folder = output_folder
+            try:
+                os.makedirs(os.path.realpath(self.output_folder))
+            except OSError:
+                pass    # file already exists, overwrite
+            self.output_path = os.path.realpath(os.path.join(self.output_folder, self.name))
+            self.bin_array = []
```

Comment:
this is a bit of a nitpick, but Python doesn't have arrays - this a list. Suggest calling it `bins` not `bin_array`.

Reply:
Renamed!